### ISSUE TITLE
LSIF and Position Encoding

### ISSUE THREAD

**Issue by Strum355** [sourcegraph/sourcegraph#15277]
**TLDR**: string encoding is hard, computers are bad
<br/>

The LSIF spec **[1]** specifies that position encoding should be based on a utf-16 encoding of documents, currently the only supported encoding assumption for positions. 

Our lsif-go indexer announces that its positions are based on a utf-16 source code **[2]**, however Go source is utf-8 based, so there is a discrepancy there **[3]**. This is probably "ok" for *us* if we also assume utf-8 on the server/client sides, so the announcement of utf-16 is ignored, while not ideal.

JS/TS source is often utf-8 as well (on various factors e.g web page encoding, most common default format for editors etc), however JS/TS strings themselves are utf-16 **[4]**. It is unclear to me whether the positions emitted by lsif-node are thus based on source codes utf-8 or the internal utf-16 representation. 

A lot of this applies to other indexers as well, and while so far we havent actually come across any issues, we dont want to be hit by surprise encoding-related positioning issues, so having a solid understanding of the state of this would be good both internally and for the wider LSIF community. Perhaps adding some info on this to lsif.dev?

cc @sourcegraph/code-intel 

[1] https://microsoft.github.io/language-server-protocol/specifications/lsif/0.4.0/specification/#metaData
[2] https://sourcegraph.com/github.com/sourcegraph/lsif-protocol/-/blob/metadata.go#L5:27
[3] https://sourcegraph.slack.com/archives/CHXHX7XAS/p1604325848476000?thread_ts=1604319725.475600&cid=CHXHX7XAS
[4] https://es5.github.io/x2.html#x2

**Comment by Strum355** [comment]
@macraig I dont think we'll be addressing this in the near future.

**Comment by macraig** [comment]
Closing, will reopen when revisited.
